lighter/callbacks/logger.py

```diff
@@ -0,0 +1,375 @@
+            metrics = getattr(pl_module, f"{mode}_metrics")
+            if metrics is not None:
+                # Compute the epoch metrics.
+                outputs["metrics"] = metrics.compute()
+                # Reset the metrics for the next epoch.
+                metrics.reset()
+
+            # Log. Done only on rank 0.
+            self._log(outputs, mode, is_epoch=True, global_step=self._get_global_step(trainer))
+
+    def _get_global_step(self, trainer: Trainer) -> int:
+        """Return the global step for the current mode. Note that when Trainer
+        is running Trainer.fit() and is in `val` mode, this method will return
+        the global step of the `train` mode in order to correctly log the validation
+        steps against training steps.
+
+        Args:
+            trainer (Trainer): Trainer, passed automatically by PyTorch Lightning.
+
+        Returns:
+            int: global step.
+        """
+        mode = LIGHTNING_TO_LIGHTER_STAGE[trainer.state.stage]
+        # When validating in Trainer.fit(), return the train steps instead of the
+        # val steps to correctly 
+        if mode == "val" and trainer.state.fn == "fit":
+            return self.global_step_counter["train"]
+        return self.global_step_counter[mode]
+
+    def on_train_epoch_start(self, trainer: Trainer, pl_module: LighterSystem) -> None:
+        # Reset the loss and the epoch step counter for the next epoch.
+        self.loss["train"] = 0
+        self.epoch_step_counter["train"] = 0
+
+    def on_validation_epoch_start(self, trainer: Trainer, pl_module: LighterSystem) -> None:
+        # Reset the loss and the epoch step counter for the next epoch.
+        self.loss["val"] = 0
+        self.epoch_step_counter["val"] = 0
+
+    def on_train_batch_end(self, trainer: Trainer, pl_module: LighterSystem, outputs: Any,
+                           batch: Any, batch_idx: int) -> None:
+        self._on_batch_end(outputs, trainer)
+
+    def on_validation_batch_end(self, trainer: Trainer, pl_module: LighterSystem, outputs: Any,
+                                batch: Any, batch_idx: int, dataloader_idx: int) -> None:
+        self._on_batch_end(outputs, trainer)
+
+    def on_test_batch_end(self, trainer: Trainer, pl_module: LighterSystem, outputs: Any,
+                          batch: Any, batch_idx: int, dataloader_idx: int) -> None:
+        self._on_batch_end(outputs, trainer)
+
+    def on_train_epoch_end(self, trainer: Trainer, pl_module: LighterSystem) -> None:
+        self._on_epoch_end(trainer, pl_module)
+
+    def on_validation_epoch_end(self, trainer: Trainer, pl_module: LighterSystem) -> None:
+        self._on_epoch_end(trainer, pl_module)
+
+    def on_test_epoch_end(self, trainer: Trainer, pl_module: LighterSystem) -> None:
+        self._on_epoch_end(trainer, pl_module)
+
+
+def preprocess_image(image: torch.Tensor) -> torch.Tensor:
+    """Preprocess the image before logging it. If it is a batch of multiple images,
```

Comment:
The logger actually does this now when passed with `log_input_as: image_N` where `N` is the number of images to be pushed. 

@ibro45 It might be a bit hard to explain this feature to general users in this format. Could we have something like `max_samples` that applies to input, target and pred together? Ties in with the `input_type` comment.

Reply:
Agree with max_samples, and it can be used for other data types too, not only images. Also agree that `image_N` might be unnecessarily complicated.

Do you think you'd want to have `max_samples` different for input, target, and pred? I can't think of a case where I would, but not sure.

Reply:
I can't think of any to be honest, especially when logging, you'd want all to be consistent I'd think.

Reply:
Implemented, it cleaned up the code nicely too